src/charm.py

```diff
@@ -288,199 +305,34 @@ def _on_remove(self, event: RemoveEvent):
-            logger.warning("PKI CSR not found in secrets")
-            return None
-        for assigned_certificate in assigned_certificates:
-            if assigned_certificate.csr == pki_csr:
-                return assigned_certificate.certificate
-        logger.info("No certificate matches the PKI CSR in secrets")
-        return None
-
-    def _on_tls_certificate_pki_certificate_available(self, event: CertificateAvailableEvent):
-        """Handle the tls-certificates-pki certificate available event."""
-        self._add_ca_certificate_to_pki_secrets_engine()
-
-    def _on_vault_pki_certificate_creation_request(
-        self, event: CertificateCreationRequestEvent
-    ) -> None:
-        """Handle the vault-pki certificate creation request event."""
-        self._generate_pki_certificate_for_requirer(
-            event.certificate_signing_request, event.relation_id
-        )
-
-    def _generate_pki_certificate_for_requirer(self, csr: str, relation_id: int):
-        """Generate a PKI certificate for a TLS requirer."""
-        if not self.unit.is_leader():
-            logger.debug("Only leader unit can handle a vault-pki certificate request")
-            return
-        if not self._tls_certificates_pki_relation_created():
-            logger.debug("TLS Certificates PKI relation not created")
-            return
-        vault = self._get_initialized_vault_client()
-        if not vault:
-            logger.debug("Failed to get initialized Vault")
-            return
-        common_name = self._get_config_common_name()
-        if not common_name:
-            logger.error("Common name is not set in the charm config")
-            return
-        if not vault.is_pki_role_created(role=PKI_ROLE, mount=PKI_MOUNT):
-            logger.debug("PKI role not created")
-            return
-        requested_csr = csr
-        requested_common_name = get_common_name_from_csr(requested_csr)
-        certificate = vault.sign_pki_certificate_signing_request(
-            mount=PKI_MOUNT,
-            role=PKI_ROLE,
-            csr=requested_csr,
-            common_name=requested_common_name,
-        )
-        if not certificate:
-            logger.debug("Failed to sign the certificate")
-            return
-        self.vault_pki.set_relation_certificate(
-            relation_id=relation_id,
-            certificate=certificate.certificate,
-            certificate_signing_request=csr,
-            ca=certificate.ca,
-            chain=certificate.chain,
+        vault.configure_policy(policy_name=CHARM_POLICY_NAME, policy_path=CHARM_POLICY_PATH)
+        cidrs = [f"{self._bind_address}/24"]
+        role_id = vault.configure_approle(
+            role_name="charm",
+            cidrs=cidrs,
+            policies=[CHARM_POLICY_NAME, "default"],
         )
+        secret_id = vault.generate_role_secret_id(name="charm", cidrs=cidrs)
+        self._set_approle_auth_secret(role_id, secret_id)
+        self.on.config_changed.emit()
```

Comment:
Why calling this here, this doesn't feel necessary

Reply:
Dan't right probably, removed it.